homeassistant/components/bring/todo.py

```diff
@@ -0,0 +1,186 @@
+        async_add_entities(
+            [
+                BringTodoListEntity(
+                    coordinator,
+                    name=f"Bring! {_list['name']}",
+                    unique_id=f"{unique_id} {_list['listUuid']}",
+                )
+            ],
+        )
+
+
+class BringTodoListEntity(
+    CoordinatorEntity[BringDataUpdateCoordinator], TodoListEntity
+):
+    """A To-do List representation of the Bring! Shopping List."""
+
+    _attr_has_entity_name = True
+    _attr_supported_features = (
+        TodoListEntityFeature.CREATE_TODO_ITEM
+        | TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.DELETE_TODO_ITEM
+        | TodoListEntityFeature.SET_DESCRIPTION_ON_ITEM
+    )
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        coordinator: BringDataUpdateCoordinator,
+        name: str,
+        unique_id: str,
+    ) -> None:
+        """Initialize BringTodoListEntity."""
+        super().__init__(coordinator)
+        self._attr_icon = "mdi:cart"
+        self._attr_name = name
+        self._attr_unique_id = unique_id
+
+    async def async_added_to_hass(self) -> None:
+        """Populate the todo list at start."""
+        self._populate_list()
+        await super().async_added_to_hass()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle the state update and prepare the todo list items."""
+        self._populate_list()
+        return super()._handle_coordinator_update()
+
+    @property
+    def list_(self) -> BringList | None:
+        """Return the bring list."""
+        return next(
+            (
+                _list
+                for _list in self.coordinator.data
+                if _list["listUuid"] == self.unique_id
+            ),
+            None,
+        )
+
+    def _populate_list(self) -> None:
+        """Populate the todo list using the coordinator data."""
+
+        _list = self.list_
+        if _list is not None:
+            self._attr_todo_items = [
+                TodoItem(
+                    uid=item["name"],
+                    summary=item["name"],
+                    description=item["specification"] or "",
+                    status=TodoItemStatus.NEEDS_ACTION,
+                )
+                for item in _list["items"]
+            ]
+        else:
+            self._attr_todo_items = None
+
+    async def async_create_todo_item(self, item: TodoItem) -> None:
+        """Add an item to the To-do list."""
+        try:
+
+            def _sync():
+                return self.coordinator.bring.saveItem(
+                    self.unique_id, item.summary, item.description or ""
+                )
+
+            await self.hass.async_add_executor_job(_sync)
+        except BringRequestException as e:
+            _LOGGER.warning("Unable to save todo item for bring")
+            raise HomeAssistantError from e
+
+        await self.async_update_ha_state(force_refresh=True)
```

Comment:
Can't you maybe ask the coördinator to refresh instead?

Reply:
You mean `await self.coordinator.async_update_ha_state(force_refresh=True)`?

Reply:
no `await self.coordinator.async_refresh()` or something like that

Reply:
And?

Reply:
Did it, should work

Reply:
Did not work, list does not get refreshed!

Reply:
I don't know why, though, as ourgroceries does it like that...